data/settings.js

```diff
@@ -10453,4 +10453,451 @@ If enabled, protocol-level SSL errors are logged. Same as
 If enabled, ignore version mismatches between different Dovecot versions.`
 	},
 
+	passdb_ldap_bind: {
+		tags: [ 'ldap' ],
+		default: 'no',
+		values: setting_types.BOOLEAN,
+		text: `
+::: info
+	LDAP Authentication Only
+:::
+
+Set \`yes\` to use authentication binding for verifying password's validity.
+
+This works by logging into LDAP server using the username and password given by client.
+
+The [\`ldap_filter\`](#pass-filter) is used to find the DN for the user.
+Note that the \`fields\` block is still used, only the password field
+is ignored in it.
+
+Before doing any search, the binding is switched back to the default DN.
+
+If you use this setting, it's a good idea to use a different
+\`ldap_connection_group\` for userdb. That way one connection is used
+only for LDAP binds and another connection is used for user lookups.
+Otherwise the binding is changed to the default DN before each user lookup.
+
+::: info Note
+If you're not using authentication binding, you'll need to give dovecot-auth
+(the user which is specified with \`ldap_auth_dn\` parameter) read access to
+\`userPassword\` field in the LDAP server.
+
+An example of this: assuming that the user assigned to \`ldap_auth_dn\` is
+"cn=authuser,dc=test,dc=dovecot,dc=net":
+
+1. Create below text file and save it as \`authuser_modify.ldif\`.
+
+::: code-group
+\`\`\`[authuser_modify.ldif]
+dn: olcDatabase={2}hdb,cn=config
+changetype: modify
+replace: olcAccess
+olcAccess: {0}to attrs=userPassword
+	by self write
+	by dn="cn=authuser,dc=test,dc=dovecot,dc=net" read
+	by * auth
+olcAccess: {1}to *
+	by self read
+	by dn="cn=authuser,dc=test,dc=dovecot,dc=net" read
+	by * auth
+\`\`\`
+:::
+
+2. Run \`ldapmodify\` to apply the change.
+
+\`\`\`console
+$ ldapmodify -Q -Y EXTERNAL -H ldapi:/// -f doveauth_access.ldif
+\`\`\``
+	},
+
+	passdb_ldap_bind_userdn: {
+		tags: [ 'ldap' ],
+		default: '<empty>',
+		values: setting_types.STRING,
+		text: `
+::: info
+	LDAP Authentication Only
+:::
+
+If authentication binding is used, you can save one LDAP request per login
+if users' DN can be specified with a common template. The template can use
+the standard %variables (see [\`user_filter\`](#user-filter)).
+
+Note that you can't use any \`fields\` declaration if you use this setting.
+
+Example: \`passdb_ldap_bind_userdn = cn=%u,ou=people,o=org\``
+	},
+
+	ldap_base: {
+		tags: [ 'ldap' ],
+		default: '<empty>',
+		values: setting_types.STRING,
+		text: `
+LDAP base.
+
+[[variable]] can be used.
+
+Example: \`ldap_base = dc=mail, dc=example, dc=org\``
+	},
+
+	use_worker: {
```

Comment:
I see this is gone eventually, but is it useful even in this intermediate commit?

Reply:
I guess these intermediates are fine doesn't really matter.